### DISCUSSION TITLE
Is it possible to disable `Lenis.onVirtualScroll` while holding down the `Shift` key?

### DISCUSSION THREAD

**Discussion by AleksandrIvanenko** [darkroomengineering/lenis#358] [Q&A] [answered]
For now `Shift` + mouse wheel leads to vertical page scrolling instead of horizontal. Can this be changed by replacing [this line](https://github.com/darkroomengineering/lenis/blob/09889bb14ad799d0f6b0760ba77184393afe3b7c/dist/lenis.js#L338) with this code: `if (event.ctrlKey || event.shiftKey)`?
Are there any reasons why this has not been done?

**Comment by clementroche** [comment]
@AleksandrIvanenko Are you sure that holding Shift while scrolling should scroll horizontally on every browser ? Here is a test: https://codepen.io/ClementRoche/pen/NWVmzNm. For instance it's not the default behaviour of MacOS touchpad.

In this case it would be `if(event.shiftKey && !this.isHorizontal && this.options.gestureOrientation === 'vertical') return`

**Reply by AleksandrIvanenko** [reply]
@clementroche, thank you for the answer!
I can't check on a Mac because I don't have one. But on Windows in Chrome, Firefox and Safari (via Playwright) Shift + mouse wheel leads to horizontal scrolling, yes.
Your suggestion seems more reliable.

**Comment by Serator** [comment]
Some more information on the different browsers.

* Chrome: https://support.google.com/chrome/answer/157179?hl=en&co=GENIE.Platform%3DDesktop#zippy=%2Ctab-window-shortcuts%2Cgoogle-chrome-feature-shortcuts%2Cwebpage-shortcuts%2Cmouse-shortcuts%2Caddress-bar-shortcuts:~:text=End-,Scroll%20horizontally%20on%20the%20page,-Shift%20%2B%20Scroll

<img width="715" alt="image" src="https://github.com/darkroomengineering/lenis/assets/625005/e8bdc8ac-6815-4ee9-8ce9-aba8f6819ce2">

* Firefox: https://support.mozilla.org/en-US/kb/mouse-shortcuts-perform-common-tasks#:~:text=Scroll-,Right,-Shift

<img width="917" alt="image" src="https://github.com/darkroomengineering/lenis/assets/625005/4f11a29c-5ee3-4f09-a918-867fc5abd627">

In addition, as far as I know this only applies to mouses, not trackpads and similar devices.

**Reply by clementroche** [reply]
That's the issues tho. As far as i know, this is not possible to detect if the user is scrolling from a touchpad or a mouse so any fix would be not reliable.

**Reply by AleksandrIvanenko** [reply]
Then maybe it's possible to give access to this virtual scroll interrupt externally (Lenis instance options?)? This way we can customize it without changing the source code. And responsibility for potential bugs will lie with us.

**Reply by clementroche** [reply]
That could be a solution.

**Reply by Serator** [reply]
> In addition, as far as I know this only applies to mouses, not trackpads and similar devices.

On trackpads with the Shift key held down, there is simply nothing happening when scrolling, it continues vertically.

**Comment by clementroche** [comment, marked as answer]
hey @AleksandrIvanenko @Serator, i've just shipped a `virtualScroll` option that allows you to manually modify the events before they get consumed. If `false` is returned, the scroll will not be smoothed. Examples: `(e) => { e.deltaY /= 2 }` (to slow down vertical scroll) or `({ event }) => !event.shiftKey` (to prevent scroll to be smoothed if shift key is pressed)

**Reply by AleksandrIvanenko** [reply]
@clementroche, looks good to me, works as expected. Thank you!
